src/main/tv/codely/scala_intro_examples/lesson_10_enums_vs_sealed_structures/exercise_solutions/jordi_puig/EatableThingsSealedType.scala

```diff
@@ -0,0 +1,24 @@
+package tv.codely.scala_intro_examples.lesson_10_enums_vs_sealed_structures.exercise_solutions.jordi_puig
+
+object EatableThingsSealedType {
+  val poisoningThreshold = 7000
+  def unapply(arg: EatableThingsSealedType): Option[(Int, Int, Int)] =
+    Some((arg.minCalories, arg.maxCalories, arg.minHeartyLevel))
+}
+
+sealed abstract class EatableThingsSealedType(
+                                            val minCalories: Int,
+                                            val maxCalories: Int,
+                                            val minHeartyLevel: Int,
+                                            val hasChocolate: Boolean = false
+                                          ) extends Ordered[EatableThingsSealedType] {
+  override def compare(that: EatableThingsSealedType): Int = this.minHeartyLevel - that.minHeartyLevel
```

Comment:
There's no need to specify `this.` while refering to the current instance attributes.